### PR TITLE
Autotip + documentation + Minor Mixin fixes.

### PR DESCRIPTION
**Pull request by Eladkay**
I've been playing with this for about a month now, and thanks to using the original Autotip Network and thanks to help I have gotten from the Autotip dev, it is fully stable.

### CODE REVIEW

**Review by palechip — COMMENTED** [review]
My first impression is that this is a major improvement. 👍 

Some questions about mixin:
1. Is Mixin only used for the development environment or will it be active in the release?
2. Are there more issues currently addressed by the Mixin addition than the resource pack issue? That one doesn't seem to apply to Minecraft 1.11.2.
3. Could these issues potentially be caused by us using wrong mappings for Minecraft? (I believe we are using 1.8.8 mappings, the correct mappings would be "stable_20")

I will continue the review and try running it once these questions are answered.

**Comment by Eladkay** [comment]
1. Right now it is not used for production needs, however there is a very strong intention to use it for such needs.
2. Not right now. There used to be another Mixin to solve Chunk-caused log spam but it was removed.
3. The Mixin issues, possibly, but unlikely.

File: src/main/java/eladkay/hudpixel/util/autotip/command/AutotipCommand.kt [hunk 1]
```diff
@@ -79,7 +78,7 @@ class AutotipCommand : HpCommandBase() {
                             "year", "yearly" -> Stats.printBetween("01-01-" + Year.now().value,
                                     "31-12-" + Year.now().value)
                             "all", "total", "life", "lifetime" -> Stats.printBetween("25-06-2016", FileUtil.date)
```
Comment (YurBoiRene) [inline src/main/java/eladkay/hudpixel/util/autotip/command/AutotipCommand.kt:80]: What is the "25-06-2016" for? Why not just log when they started or at the beginning?

File: src/main/java/eladkay/hudpixel/util/autotip/command/AutotipCommand.kt [hunk 2]
```diff
@@ -89,41 +88,40 @@ class AutotipCommand : HpCommandBase() {
                     Autotip.toggle = !Autotip.toggle
                     ClientMessage.send(
                             "Autotipper: " + (if (Autotip.toggle)
-                                ChatColor.GREEN.toString() + "En"
+                                EnumChatFormatting.GREEN.toString() + "En"
                             else
-                                ChatColor.RED.toString() + "Dis") + "abled")
+                                EnumChatFormatting.RED.toString() + "Dis") + "abled")
```
Comment (YurBoiRene) [inline src/main/java/eladkay/hudpixel/util/autotip/command/AutotipCommand.kt:93]: You can just use + to combine chat colors with strings.

File: src/main/java/eladkay/hudpixel/util/autotip/command/TipHistoryCommand.kt [hunk 3]
```diff
@@ -43,29 +40,64 @@ class TipHistoryCommand : HpCommandBase() {
...
 
             if (page < 1 || page > pages) {
-                ClientMessage.send(ChatColor.RED.toString() + "Invalid page number.")
+                ClientMessage.send(EnumChatFormatting.RED.toString() + "Invalid page number.")
             } else {
                 ClientMessage.separator()
-                ClientMessage.send(ChatColor.GOLD.toString() + "Tip History " + ChatColor.GRAY
-                        + "[Page " + page + " of " + pages + "]" + ChatColor.GOLD + ":")
+                ClientMessage.send(EnumChatFormatting.GOLD.toString() + "Tip History " + EnumChatFormatting.GRAY
+                        + "[Page " + page + " of " + pages + "]" + EnumChatFormatting.GOLD + ":")
```
Comment (YurBoiRene) [inline src/main/java/eladkay/hudpixel/util/autotip/command/TipHistoryCommand.kt:47]: Why do you use .toString() and sometimes just +?

**Review by YurBoiRene — APPROVED** [review]
Looks all good to me except for some minor things

File: src/main/java/eladkay/hudpixel/util/autotip/command/AutotipCommand.kt (hunk 1 above)
Comment (Eladkay) [reply src/main/java/eladkay/hudpixel/util/autotip/command/AutotipCommand.kt:80]: It’d require us to track another thing and we don’t need that

File: src/main/java/eladkay/hudpixel/util/autotip/command/TipHistoryCommand.kt (hunk 3 above)
Comment (Eladkay) [reply src/main/java/eladkay/hudpixel/util/autotip/command/TipHistoryCommand.kt:47]: It’s adapted code from the original auto tip that has the toString.

File: src/main/java/eladkay/hudpixel/util/autotip/command/AutotipCommand.kt (hunk 2 above)
Comment (Eladkay) [reply src/main/java/eladkay/hudpixel/util/autotip/command/AutotipCommand.kt:93]: I know, I wrote that method.

**Review by unaussprechlich — CHANGES_REQUESTED** [review]
- [ ] Make a option to Hide the tip messages!
- [ ] update the booster display when a gm got tiped
- [ ] keep track of the XP they earned. 
- [ ] (Optional) If you do the client side I would like to store the amount in a database and also keep track of xp they earned there. Some cool commands like "/xpearned DATE ANOTHER_DATE" would make is so all those XP nerds can keep track of their earnings. #17

**Comment by Eladkay** [comment]
1. I already do, see toggles and MessageOption.
2. It should, if not that’s your thing and I’m not touching it.
3. I do. Try /hudat stats
4. See 3

**Review by unaussprechlich — APPROVED** [review]

**Pull request merged**